pkg/plugin/aggregation/run_test.go

```diff
@@ -182,95 +183,194 @@ func TestRunAndMonitorPlugin(t *testing.T) {
 	}
 }
 
-func getTestCert() (*tls.Certificate, error) {
-	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		return nil, errors.Wrap(err, "couldn't generate private key")
-	}
-	tmpl := &x509.Certificate{
-		SerialNumber: big.NewInt(0),
-	}
-	certDER, err := x509.CreateCertificate(rand.Reader, tmpl, tmpl, &privKey.PublicKey, privKey)
-	if err != nil {
-		return nil, errors.Wrap(err, "couldn't create certificate")
-	}
-
-	return &tls.Certificate{
-		Certificate: [][]byte{certDER},
-		PrivateKey:  privKey,
-	}, nil
-}
-
-type MockCleanupPlugin struct {
-	skipCleanup bool
-	cleanedUp   bool
-}
-
-func (cp *MockCleanupPlugin) Run(_ kubernetes.Interface, _ string, _ *tls.Certificate, _ *corev1.Pod, _, _ string) error {
+func (cp *MockRunPlugin) Run(_ kubernetes.Interface, _ string, _ *tls.Certificate, _ *corev1.Pod, _, _ string) error {
 	return nil
 }
 
-func (cp *MockCleanupPlugin) Cleanup(_ kubernetes.Interface) {
+func (cp *MockRunPlugin) Cleanup(_ kubernetes.Interface) {
 	cp.cleanedUp = true
 }
 
-func (cp *MockCleanupPlugin) Monitor(_ context.Context, _ kubernetes.Interface, _ []corev1.Node, _ chan<- *plugin.Result) {
+func (cp *MockRunPlugin) Monitor(_ context.Context, _ kubernetes.Interface, _ []corev1.Node, _ chan<- *plugin.Result) {
 	return
 }
 
-func (cp *MockCleanupPlugin) ExpectedResults(_ []corev1.Node) []plugin.ExpectedResult {
+func (cp *MockRunPlugin) ExpectedResults(_ []corev1.Node) []plugin.ExpectedResult {
 	return []plugin.ExpectedResult{}
 }
 
-func (cp *MockCleanupPlugin) FillTemplate(_ string, _ *tls.Certificate) ([]byte, error) {
+func (cp *MockRunPlugin) FillTemplate(_ string, _ *tls.Certificate) ([]byte, error) {
 	return []byte{}, nil
 }
 
-func (cp *MockCleanupPlugin) GetName() string {
-	return "mock-cleanup-plugin"
+func (cp *MockRunPlugin) GetName() string {
+	return cp.name
 }
 
-func (cp *MockCleanupPlugin) SkipCleanup() bool {
+func (cp *MockRunPlugin) SkipCleanup() bool {
 	return cp.skipCleanup
 }
 
-func (cp *MockCleanupPlugin) GetResultFormat() string {
+func (cp *MockRunPlugin) GetResultFormat() string {
 	return ""
 }
 
-func (cp *MockCleanupPlugin) GetResultFiles() []string {
+func (cp *MockRunPlugin) GetResultFiles() []string {
 	return []string{}
 }
 
-func (cp *MockCleanupPlugin) GetDescription() string {
+func (cp *MockRunPlugin) GetDescription() string {
 	return "A mock plugin used for testing purposes"
 }
 
-func (cp *MockCleanupPlugin) GetSourceURL() string {
+func (cp *MockRunPlugin) GetSourceURL() string {
 	return ""
 }
 
+func (cp *MockRunPlugin) GetOrder() int {
+	return cp.order
+}
+
+type MockRunPlugin struct {
+	skipCleanup bool
+	cleanedUp   bool
+	order       int
+	name        string
+}
+
+func TestGetOrderedPlugins(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		input    []plugin.Interface
+		expected [][]plugin.Interface
+	}{
+		{
+			desc: "Ensure lists of plugin.Interface are ordered by plugin order",
```

Comment:
nit: Very simple to add but be sure to include empty slice and null case, just to make sure we pin that behavior down and a change with that logic wont break something else.